### DISCUSSION TITLE
Is the Sirius Web is build  with the GLSP ?

### DISCUSSION THREAD

**Discussion by lak-proddev** [eclipse-sirius/sirius-components#3115] [Q&A] [answered]
Is the Sirius Web is built with the GLSP? Can you share the architecture of Sirius Web?

**Comment by pcdavid** [comment, marked as answer]
No we don't use GLSP, which is only suited for diagrams.

Sirius Web is designed to support all kinds of graphical representations, diagrams are just one example. For example in Sirius Web we also support forms, trees (the explorer), and now Gantt, Deck (Trello-like boards), etc. 

We use GraphQL as the base protocol between the front and back, which offers the flexibility we need.

Have a look at https://github.com/eclipse-sirius/sirius-web/tree/master/doc/reference to get an overview of the architecture, and search for the `*.graphqls` files for the details of the GraphQL Schema used to interact with each kind of representation (or better yet: connect to a live instance at `/api/graphql` with a GraphQL client and discover the schema interactively).

**Reply by lak-proddev** [reply]
Thank you very much for the quick reply.
